google-api-go-generator/gen.go

```diff
@@ -498,6 +498,42 @@ func (a *API) apiBaseURL() string {
 	return resolveRelative(base, rel)
 }
 
+// apiBaseURLTemplate returns the value returned by apiBaseURL with the
+// following changes: 1) The hostname (excluding subdomains) is replaced with %s
+// for universe domain substitution. 2) If the value does not have a scheme,
+// https:// is prepended.
+func (a *API) apiBaseURLTemplate() (string, error) {
+	baseURL := a.apiBaseURL()
+	return replaceDomain(baseURL, "%s")
```

Comment:
Should we use better sentinel value for the template than "%s"?  My worry is that might be ambigous.

Have you seen any discussions for how APIs might produce the template form directly (via a field in the discovery doc or the API yaml configs)?  This is a reason I didn't consider the template option initially is this is a resolution issue, and so we're simply splitting the resolution logic up between here and elsewhere (e.g. google.golang.org/api/internal).

Reply:
> Should we use better sentinel value for the template than "%s"?

I'm totally open to this. I can update the PR with whatever string seems most common across languages.

> we're simply splitting the resolution logic up between here and elsewhere

I considered this as well. My conclusion was that it's important to make the replacement mechanism explicit to users by configuring the template and the default universe domain in the client setup code, even at the cost of duplicating the messy work of creating the template string. By moving this upstream to the generator, things are going to be clear(er) for users who inspect the source of their clients. Dynamically changing the endpoint passed in `WithDefaultEndpoint` after client init seems much harder for users to comprehend.

Reply:
I changed the sentinel value from `%s` to `UNIVERSE_DOMAIN`, thanks!